File: selectionSort.py
def selection_sort(my_list):
    for i in range(len(my_list)):
        minimum = i 
        print(minimum)
        for j in range(i+1, len(my_list)):
            if my_list[minimum] > my_list[j]:
                minimum = j
        # swap
        my_list[i], my_list[minimum] = my_list[minimum], my_list[i]

my_list = [64, 25, 12, 22, 11]
selection_sort(my_list)
print("Sorted array:", my_list)

File: selectionSort.java
public class SelectionSort {
    public static void selectionSort(int[] A) {
        int n = A.length;
        for (int i = 0; i < n - 1; i++) {
            int minimum = i;
            for (int j = i + 1; j < n; j++) {
                if (A[j] < A[minimum]) {
                    minimum = j;
                }
            }
            int temp = A[i];
            A[i] = A[minimum];
            A[minimum] = temp;
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        selectionSort(arr);
        System.out.print("Sorted array: ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
